src/engine/docker_engine.rs

```diff
@@ -43,6 +43,35 @@ impl DockerEngine {
 
         let config = ApplicationConfig::from_env();
 
+        // Create the network if needed
+        let mut filters: HashMap<&str, Vec<&str>> = HashMap::new();
+        filters.insert("name", vec![&config.traefik_config.network_name]);
+        let list_options: ListNetworksOptions<&str> = ListNetworksOptions { filters };
```

Comment:
Is it possible to do something like that instead? (idk if that compiles)

```suggestion
        let list_options: ListNetworksOptions<&str> = ListNetworksOptions {
            filters: HashMap::from([
                ("name", vec![&config.traefik_config.network_name]),
            ]),
        };
```

Reply:
No it doesn't compile, I tried this at first then decided to follow the documentation. 
Using as_ref() instead of & did the trick though